babar-agent/src/main/java/com/criteo/babar/agent/profiler/utils/ProcFSUtils.java

```diff
@@ -174,7 +174,55 @@ protected static ProcPidIO parsePidIO(String output) {
         return new ProcPidIO(rchar, wchar, readBytes, writeBytes);
     }
 
-    public static class ProcPidStat {
+    public static ProcNetIO netio() throws IOException {
+        String outputProcNetDev = OSUtils.exec(new String[]{"cat", "/proc/net/dev"});
+        String outputSysClassNet = OSUtils.exec(new String[]{"ls", "-l", "/sys/class/net"});
+        return parseNetIO(outputProcNetDev, outputSysClassNet);
+    }
+
+    protected static ProcNetIO parseNetIO(String outputProcNetDev, String outputSysClassNet) {
+    	List<String> realDevices = parseRealDevices(outputSysClassNet);
+    	String[] lines = outputProcNetDev.split("\n");
+        if (!lines[0].startsWith("Inter-|")) {
+            throw new RuntimeException("Unexpected first line in output of /proc/dev/net:\n" + outputProcNetDev);
+        }
+
+        if (lines.length < 2) {
+            throw new RuntimeException("Not enough lines in output of /proc/dev/net");
+        }
+
+        long rxBytes = 0;
+        long txBytes = 0;
+
+        for(int i=2; i < lines.length; i++) {
+            String[] cols = lines[i].trim().split(" +"); //Trim leading spaces
+
+            if(cols.length == 17 && realDevices.contains(cols[0])) {
+                rxBytes += Long.parseLong(cols[1]);
+                txBytes += Long.parseLong(cols[9]);
+            }
+        }
+
+        return new ProcNetIO(rxBytes, txBytes);
+	}
+
+	protected static List<String> parseRealDevices(String outputSysClassNet) {
+		List<String> realDevices = new LinkedList<>();
+		String[] lines = outputSysClassNet.split("\n");
+		
+		for(String line : lines) {
+			String[] cols = line.split(" +");
+			
+			if(cols.length == 11) {
+				if(!cols[10].startsWith("../../devices/virtual/net/")) {
```

Comment:
the two `if`s could be factorized as
`if (cols.length == 11 && !cols[10].startsWith("../../devices/virtual/net/")`